Ports: cannot build gmake 4.2/4.2.1

When cross-compiling from Arch Linux, building GNU Make fails due to the presence of the flag `-pthread`. The full log is [here](https://www.sgregoratto.me/paste/fail.log) and the relevant parts are below:

```
make[2]: Entering directory '/home/stephen/gits/serenity/Ports/make/make-4.2'
i686-pc-serenity-gcc -DLOCALEDIR=\"/usr/local/share/locale\" -DLIBDIR=\"/usr/local/lib\" -DINCLUDEDIR=\"/usr/local/include\" -DHAVE_CONFIG_H -I.  -I./glob   -I/usr/include/guile/2.0 -pthread  -g -O2 -MT ar.o -MD -MP -MF .deps/ar.Tpo -c -o ar.o ar.c
i686-pc-serenity-gcc -DLOCALEDIR=\"/usr/local/share/locale\" -DLIBDIR=\"/usr/local/lib\" -DINCLUDEDIR=\"/usr/local/include\" -DHAVE_CONFIG_H -I.  -I./glob   -I/usr/include/guile/2.0 -pthread  -g -O2 -MT arscan.o -MD -MP -MF .deps/arscan.Tpo -c -o arscan.o arscan.c
i686-pc-serenity-gcc -DLOCALEDIR=\"/usr/local/share/locale\" -DLIBDIR=\"/usr/local/lib\" -DINCLUDEDIR=\"/usr/local/include\" -DHAVE_CONFIG_H -I.  -I./glob   -I/usr/include/guile/2.0 -pthread  -g -O2 -MT commands.o -MD -MP -MF .deps/commands.Tpo -c -o commands.o commands.c
i686-pc-serenity-gcc -DLOCALEDIR=\"/usr/local/share/locale\" -DLIBDIR=\"/usr/local/lib\" -DINCLUDEDIR=\"/usr/local/include\" -DHAVE_CONFIG_H -I.  -I./glob   -I/usr/include/guile/2.0 -pthread  -g -O2 -MT default.o -MD -MP -MF .deps/default.Tpo -c -o default.o default.c
i686-pc-serenity-gcc -DLOCALEDIR=\"/usr/local/share/locale\" -DLIBDIR=\"/usr/local/lib\" -DINCLUDEDIR=\"/usr/local/include\" -DHAVE_CONFIG_H -I.  -I./glob   -I/usr/include/guile/2.0 -pthread  -g -O2 -MT dir.o -MD -MP -MF .deps/dir.Tpo -c -o dir.o dir.c
i686-pc-serenity-gcc -DLOCALEDIR=\"/usr/local/share/locale\" -DLIBDIR=\"/usr/local/lib\" -DINCLUDEDIR=\"/usr/local/include\" -DHAVE_CONFIG_H -I.  -I./glob   -I/usr/include/guile/2.0 -pthread  -g -O2 -MT expand.o -MD -MP -MF .deps/expand.Tpo -c -o expand.o expand.c
i686-pc-serenity-gcc -DLOCALEDIR=\"/usr/local/share/locale\" -DLIBDIR=\"/usr/local/lib\" -DINCLUDEDIR=\"/usr/local/include\" -DHAVE_CONFIG_H -I.  -I./glob   -I/usr/include/guile/2.0 -pthread  -g -O2 -MT file.o -MD -MP -MF .deps/file.Tpo -c -o file.o file.c
i686-pc-serenity-gcc -DLOCALEDIR=\"/usr/local/share/locale\" -DLIBDIR=\"/usr/local/lib\" -DINCLUDEDIR=\"/usr/local/include\" -DHAVE_CONFIG_H -I.  -I./glob   -I/usr/include/guile/2.0 -pthread  -g -O2 -MT function.o -MD -MP -MF .deps/function.Tpo -c -o function.o function.c
i686-pc-serenity-gcc: error: unrecognized command line option '-pthread'
i686-pc-serenity-gcc: error: unrecognized command line option '-pthread'
make[2]: *** [Makefile:700: ar.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [Makefile:700: arscan.o] Error 1
i686-pc-serenity-gcc: error: unrecognized command line option '-pthread'
i686-pc-serenity-gcc: error: unrecognized command line option '-pthread'
make[2]: *** [Makefile:700: commands.o] Error 1
make[2]: *** [Makefile:700: default.o] Error 1
i686-pc-serenity-gcc: error: unrecognized command line option '-pthread'
i686-pc-serenity-gcc: error: unrecognized command line option '-pthread'
make[2]: *** [Makefile:700: dir.o] Error 1
make[2]: *** [Makefile:700: file.o] Error 1
i686-pc-serenity-gcc: error: unrecognized command line option '-pthread'
make[2]: *** [Makefile:700: expand.o] Error 1
i686-pc-serenity-gcc: error: unrecognized command line option '-pthread'
make[2]: *** [Makefile:700: function.o] Error 1
make[2]: Leaving directory '/home/stephen/gits/serenity/Ports/make/make-4.2'
make[1]: *** [Makefile:798: all-recursive] Error 1
make[1]: Leaving directory '/home/stephen/gits/serenity/Ports/make/make-4.2'
make: *** [Makefile:534: all] Error 2
```

--- a/Ports/make/package.sh
+++ b/Ports/make/package.sh
@@ -1,6 +1,6 @@
 #!/bin/bash ../.port_include.sh
 port=make
-version=4.2
+version=4.2.1
 useconfigure=true
-files="https://ftp.gnu.org/gnu/make/make-4.2.tar.bz2 make-4.2.tar.bz2"
-configopts="--target=i686-pc-serenity --with-sysroot=/"
+files="https://ftp.gnu.org/gnu/make/make-4.2.1.tar.bz2 make-4.2.1.tar.bz2"
+configopts="--target=i686-pc-serenity --with-sysroot=/ --without-guile"

